tap_sailthru/discover.py

```diff
@@ -98,6 +98,15 @@ def discover(config) -> Catalog:
 
     for schema_name, schema in schemas.items():
         schema_meta = schemas_metadata[schema_name]
+        stream_obj = STREAMS[schema_name]
+        meta_map = metadata.to_map(schema_meta)
+
+        parent_attr = getattr(stream_obj, "parent", None)
+        parent_tap_stream_id = None
+        if parent_attr:
+            parent_tap_stream_id = parent_attr.__name__ if isinstance(parent_attr, type) else str(parent_attr)
+            meta_map = metadata.write(meta_map, (), 'parent-tap-stream-id', parent_tap_stream_id)
+        schema_meta = metadata.to_list(meta_map)
```

Comment:
Can we try to simplify implementation by adding helper function `_get_parent_tap_stream_id()` like below?

```suggestion
        stream_obj = STREAMS[schema_name]
  
        # Extract parent information more cleanly
        parent_tap_stream_id = _get_parent_tap_stream_id(stream_obj)
        
        # Add parent metadata if it exists
        if parent_tap_stream_id:
            meta_map = metadata.to_map(schema_meta)
            meta_map = metadata.write(meta_map, (), 'parent-tap-stream-id', parent_tap_stream_id)
            schema_meta = metadata.to_list(meta_map)
```